# --- T2-COPYRIGHT-BEGIN ---
# t2/package/*/openexr/hotfix-glibc-2.43.patch
# Copyright (C) 2026 The T2 SDE Project
# SPDX-License-Identifier: GPL-2.0 or patched project license
# --- T2-COPYRIGHT-END ---

--- openexr-3.4.4/src/lib/OpenEXRCore/internal_thread.h.vanilla
+++ openexr-3.4.4/src/lib/OpenEXRCore/internal_thread.h
@@ -29,13 +29,15 @@
 }
 #    else
 #        include <pthread.h>
-#        define ONCE_FLAG_INIT PTHREAD_ONCE_INIT
+#        ifndef ONCE_FLAG_INIT
 typedef pthread_once_t once_flag;
+#        define ONCE_FLAG_INIT PTHREAD_ONCE_INIT
 static inline void
 call_once (once_flag* flag, void (*func) (void))
 {
     (void) pthread_once (flag, func);
 }
+#        endif
 #    endif
 #else
 #    define ONCE_FLAG_INIT 0
